File: frontend/src/types/Lesson.ts
import {Student} from "./Student.ts";

export type Lesson ={
    id: string;
    name: string;
    studentList: Student[];
    attendanceList : Attendance[];
    messageList : Message[];
}
export type Attendance ={
    description: string;
    status : string;
}
export enum AttendanceStatus{
    LOW="LOW",
    MEDIUM="MEDIUM",
    HIGH="HIGH"
}
export type Message={
    message: string;
    student: Student;
}

File: backend/src/main/java/com/example/backend/model/Lesson.java
package com.example.backend.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.With;
import org.apache.logging.log4j.message.Message;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Data
@With
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "lessons")
public class Lesson {
    private String id;
    private String name;
    List<Student> studentList;
    List<AttendanceDTO> attendanceList;
    List<MessageDTO> messageList;
    public Lesson (LessonDTO lessonDTO){
        this(null, lessonDTO.getName(), lessonDTO.getStudentList(),lessonDTO.getAttendanceList(),lessonDTO.getMessageList());
    }
}
